Cargo.toml

```diff
@@ -113,19 +113,55 @@ debug = 1
 #
 # The reasons for each of these overrides are listed in deny.toml.
 [patch.crates-io]
-# IMPORTANT: you should only depend on "main", "master", or an upstream release
-# branch (e.g., "v7.x"). Do *not* depend on a feature/patch branch (e.g.,
-# "fix-thing" or "pr-1234"). Feature/patch branches tend to get rewritten or
-# disappear (e.g., because a PR is force pushed or gets merged), after which
-# point it becomes impossible to build that historical version of Materialize.
-chrono = { git = "https://github.com/chronotope/chrono.git", branch = "0.4.x" }
+# Waiting for hashlink, indexmap, and lru to upgrade to hashbrown v0.13,
```

Comment:
This commit moves most `git` dependencies from the individual project `Cargo.toml`s to the patch section here. AIUI the reason for this is to have a single place where we document these special cases and why we forked them. I like this, but I have two observations:

First, a couple lines above there is this comment:
```
# Use this section only to change the source of dependencies that appear as
# transitive dependencies of other external dependencies in the dependency
# graph. For everything else (e.g. rust-postgres, rdkafka,
# differential-dataflow, proptest, timely), use a `git` source directly in each
# Cargo.toml.
```

This goes directly against what this commit is doing. We should probably change/remove that comment. Also, what were the reasons for not putting all the git overrides in the `patch` section before?

Second, there are a couple instances left where we still specify the git overrides directly in the per-project `Cargo.toml`s. Most of them are timely/differential/columnation, but I see also `tracing-opentelemetry`, `console-subscriber` and `launchdarkly-server-sdk`. Are these somehow special or just oversights?

Reply:
> This goes directly against what this commit is doing. We should probably change/remove that comment. Also, what were the reasons for not putting all the git overrides in the patch section before?

Whoops, great catch. Fixed. The patch directive was riddled with bugs a few years ago, when this comment was written, and would occasionally just fail to kick in. So it was worth avoiding the patch directive whenever possible. Those bugs all have, anecdotally, been fixed for quite a while, so I think it's safe enough to always use the patch directive now.



> Second, there are a couple instances left where we still specify the git overrides directly in the per-project `Cargo.toml`s. Most of them are timely/differential/columnation, but I see also `tracing-opentelemetry`, `console-subscriber` and `launchdarkly-server-sdk`. Are these somehow special or just oversights?

Whoops! Just oversights. Thanks for the eagle eyes. All cleaned up now.